File: hoteltycoon-backend/src/main/java/be/pxl/hotel/service/HotelService.java
package be.pxl.hotel.service;

import java.util.List;

import be.pxl.hotel.api.request.CreateHotelRequest;
import be.pxl.hotel.api.response.HotelDTO;
import be.pxl.hotel.api.response.HotelDetailDTO;
import be.pxl.hotel.domain.Facility;
import be.pxl.hotel.domain.Hotel;

public interface HotelService {

	Long createHotel(CreateHotelRequest request);

	List<HotelDTO> getAllHotels();

	HotelDetailDTO getHotelDetails(Long hotelId);

	void buyBuilding(Long hotelId);

	void buyFacility(Long hotelId, Facility facility);

	List<Hotel> findAllOpenHotels();

	void book(Hotel hotel, int persons, int nights);
}

File: hoteltycoon-frontend/src/services/hotel.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Buildingplot} from "../models/buildingplot.model";
import {Hotel} from "../models/hotel.model";
import {HotelDetails} from "../models/hotel-details.model";
import {Facility} from "../models/facility.model";
import {environment} from "../environment";

@Injectable({
  providedIn: 'root',
})
export class HotelService {// Replace with your actual API endpoint

  constructor(private http: HttpClient) {
  }

  createHotel(hotelName: string, buildingplotId: number): Observable<any> {
    const headers = { 'Content-Type': 'application/json' };
    // Make an HTTP POST request to your backend to buy the land
    const body = {
      "name": hotelName,
      "buildingplotId": buildingplotId
    };
    return this.http.post(`${environment.backendUrl}/hotels`, body, { headers });
  }

  getHotels(): Observable<Hotel[]> {
    return this.http.get<Hotel[]>(`${environment.backendUrl}/hotels`);
  }

  getHotelDetails(hotelId: number): Observable<HotelDetails> {
    return this.http.get<HotelDetails>(`${environment.backendUrl}/hotels/` + hotelId);
  }

  buyBuilding(hotelId: number) {
    return this.http.post(`${environment.backendUrl}/hotels/` + hotelId + `/buy-building`, null);
  }

  buyFacility(hotelId: number, facility: string) {
    return this.http.post(`${environment.backendUrl}/hotels/` + hotelId + `/buy-facility/` + facility, null);
  }
}
